### DISCUSSION TITLE
Auth Related Question: Is it possible to remove a user's email from the access_token

### DISCUSSION THREAD

**Discussion by Josh-Rosen** [supabase/supabase#22276] [Questions]
I'm interested in using Supabase for Auth. I have a separate backend server I'm using, and I've already implemented checking the JWT with my Supabase HMAC secret.

I noticed the JWT access_token payload includes unencrypted PII, specifically my user's email and potential phone number.

Is it possible to produce a JWT payload without this PII data? Overall using Supabase's JWT and HMAC solutions work well, however sending unencrypted PII with the access_token seems a bit risky.

Is it possible to generate the access_token without PII? Do I need to use this method here https://github.com/supabase/auth-js/blob/b887af5567409a189b0e700880a032c65d0d21e8/src/GoTrueClient.ts#L301 to set my own token?

Here's an example of the token payload via https://jwt.io/

```
{
  "aud": "authenticated",
  "exp": 1711497688,
  "iat": 1711494088,
  "iss": "https://{my-project}.supabase.co/auth/v1",
  "sub": "{uuid}",
  "email": "email@email.com",
  "phone": "555-555-5555",
  "app_metadata": {
    "provider": "email",
    "providers": [
      "email"
    ]
  },
  "user_metadata": {
    "email": "email@email.com",
    "email_verified": false,
    "phone_verified": false,
    "sub": "{uuid}"
  },
  "role": "authenticated",
  "aal": "aal1",
  "amr": [
    {
      "method": "password",
      "timestamp": 1711217784
    }
  ],
  "session_id": "{uuid}",
  "is_anonymous": false
}
```

**Comment by GaryAustin1** [comment]
No options in general with auth and jwt's from the API calls.

There is a new feature that allows generating a custom jwt claim.  It MIGHT be able to remove a claim?
https://supabase.com/docs/guides/auth/auth-hooks

**Reply by Josh-Rosen** [reply]
Yeah, this is definitely an option, though I was hoping for something simpler.
Technically, as long as all requests are through HTTPS and the Token isn't logged anywhere then there's not much risk of leaking PII, but still it's strange to me that the token would include sensitive information.

**Comment by a3har** [comment]
For anyone coming now, the way to do this is via custom access token hooks from Supabase. 

Documentation: https://supabase.com/docs/guides/auth/auth-hooks/custom-access-token-hook
